tools/automation/cli_linter/util.py

```diff
@@ -43,26 +42,23 @@ def _filter_mods(command_table, help_file_entries, modules=None, extensions=None
         is_specified = source_name in extensions if is_extension else source_name in filtered_module_names
         if is_specified == exclude:
             # brute force method of ignoring commands from a module or extension
-            del command_table[command_name]
+            command_loader.command_table.pop(command_name, None)
             help_file_entries.pop(command_name, None)
-            for group_name in get_command_groups(command_name):
-                help_file_entries.pop(group_name, None)
 
-    return command_table, help_file_entries
+        # Remove unneeded command groups
+        retained_command_groups = set([' '.join(x.split(' ')[:-1]) for x in command_loader.command_table])
+        excluded_command_groups = set(command_loader.command_group_table.keys()) - retained_command_groups
+        for group_name in excluded_command_groups:
+            command_loader.command_group_table.pop(group_name, None)
+            help_file_entries.pop(group_name, None)
```

Comment:
This is inside a for loop over all the commands in the command_table.
Please either:
1. Get a list of group_names from the command_name in the iteration and only delete those.
or
2. Move this outside after the for loop.

I believe `1.` will save you a line or two of code and prevent the need to use set operations.

Reply:
Done